### ISSUE TITLE
Lucene Indices are broken on Windows

### ISSUE THREAD

**Issue by silvanheller** [vitrivr/cottontaildb#71]
I'll just leave this here for reference.
Windows continously causes issues (#58 #57 #7). We can leave this as wontfix or we can find a way to CI-Test cottontail releases against windows.

To reproduce, run the Unit-Tests in Cineast against a windows instance of Cottontail.

**Comment by silvanheller** [comment]
This did not solve the issue on my windows machine

**Comment by ppanopticon** [comment]
Setting `"enableMmap": false` does not help?

**Comment by silvanheller** [comment]
Yes. I've modified the ci runner on `dev` to test both windows & ubuntu with appropriate config-options (see https://github.com/vitrivr/cottontaildb/actions/runs/664468619)
As far as my machine goes: On the Cottontail side, the `LuceneIndexTest fails` and the `CottontailIntegrationtest` of Cineast fails (having deleted the data-folder before starting cottontail and with `enableMmap:false` with 

```
java.nio.file.AccessDeniedException: .\cottontaildb-data\schema_cineast\entity_test_feature_text\idx_text_lucene.lucene\_1.cfs
	at sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:89) ~[?:?]
	at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103) ~[?:?]
	at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108) ~[?:?]
	at sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:274) ~[?:?]
	at sun.nio.fs.AbstractFileSystemProvider.delete(AbstractFileSystemProvider.java:105) ~[?:?]
	at java.nio.file.Files.delete(Files.java:1141) ~[?:?]
	at org.vitrivr.cottontail.utilities.io.FileUtilities$deleteRecursively$1.accept(FileUtilities.kt:22) ~[main/:?]
	at org.vitrivr.cottontail.utilities.io.FileUtilities$deleteRecursively$1.accept(FileUtilities.kt:13) ~[main/:?]
```

**Comment by ppanopticon** [comment]
I believe I found and fixed this issue. At least the UnitTest now works on Windows 10.

The problem was simply, that I did not close the index properly when closing an entity. That obviously caused the indexes to misbehave due to dangeling file locks.

**Comment by ppanopticon** [comment]
Does the issue still occur on 0.12.5? I fixed a number of bugs and the unit tests now run through both on _Windows_ and _Ubuntu_.

Still, though, you need to switch off _memory mapped files_ by setting `"enableMmap": false when importing data or updating indexes. Read-only access should not cause an issue.

**Comment by ppanopticon** [comment]
Seems to be fixed.
